californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -445,12 +450,12 @@ public synchronized void start() throws IOException {
 		}
 
 		if (this.executor == null) {
-			LOGGER.info("Endpoint [{}] requires an executor to start, using default single-threaded daemon executor", getUri());
+			int threadCount = config.getInt(Keys.PROTOCOL_STAGE_THREAD_COUNT);
```

Comment:
Why changing the number of thread from 1 to `Keys.PROTOCOL_STAGE_THREAD_COUNT` ?

I think this may have impact on tests where we just create endpoints with default value.

For real use case, the executor is created in coapServer and so already use `Keys.PROTOCOL_STAGE_THREAD_COUNT`

Reply:
> I think this may have impact on tests where we just create endpoints with default value.

Yes, therefore the `CoapNetworkRule` adjust to number of  `PROTOCOL_STAGE_THREAD_COUNT`.

> For real use case, the executor is created in coapServer and so already use

Depends on the expectations for the endpoints used by clients.
FMPOV, using that configuration value also here, enables to configure the client.
Without, each client has to configure the endpoint explicit.
But if you have a too strange feeling (beside the tests), I can revert that to a fixed one thread.

Reply:
I think you have a better knowledge of the tests code base.
I just raise a waning, if you thinks it's ok, that's good to me.